src/components/ModalWithForm.js

```diff
@@ -1,14 +1,15 @@
-import Modal from "./Modal.js";
+import Modal from "./Modal";
 
-export default class ModalWithForm extends Modal {
-  constructor(modalSelector, handleFormSubmit) {
-    super(modalSelector);
+class ModalWithForm extends Modal {
+  constructor({ modalSelector, handleFormSubmit }) {
+    super({ modalSelector });
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
   }
 
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```